### PR TITLE
Added partial search for schools

### PR DESCRIPTION
**Pull request by edwoodward**

### CODE REVIEW

File: app/controllers/api/v1/schools_controller.rb [hunk 1]
```diff
@@ -28,4 +28,12 @@ def show
       }, status: :not_found
     end
   end
+
+  # GET schools/search?partial=schoo
+  def search
+    if params['partial'].present?
```
Comment (mwharrison) [inline app/controllers/api/v1/schools_controller.rb:34]: I don't think partial searches should be using another query param. Can you combine this with `=name`?

File: spec/requests/schools_request_spec.rb [hunk 2]
```diff
@@ -48,4 +48,14 @@
     expect(response).to have_http_status(:success)
   end
 
+  it 'returns a successful response for school by partial name' do
```
Comment (mwharrison) [inline spec/requests/schools_request_spec.rb:51]: this should be added to https://github.com/openstax/openstax-salesforce-api/blob/master/spec/integration/v1/schools_spec.rb 

Here is an example of searching in an rswag block.
https://github.com/rswag/rswag/blob/0def9fad6fbfc97168d0e1ffc740c115e00c0768/test-app/spec/integration/blogs_spec.rb#L31-L49

**Review by mwharrison — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/schools_controller.rb (hunk 1 above)
Comment (edwoodward) [reply app/controllers/api/v1/schools_controller.rb]: Moved this to use the name parameter

File: spec/requests/schools_request_spec.rb (hunk 2 above)
Comment (edwoodward) [reply spec/requests/schools_request_spec.rb]: All tests moved to Swagger test

Comment (edwoodward) [reply spec/requests/schools_request_spec.rb]: Done

**Review by mwharrison — APPROVED** [review]

**Pull request merged**